python/feathub/processors/spark/spark_processor.py

```diff
@@ -0,0 +1,130 @@
+# Copyright 2022 The Feathub Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from concurrent.futures import ThreadPoolExecutor
+from datetime import timedelta, datetime
+from typing import Union, Optional, Dict
+
+import pandas as pd
+from pyspark.sql import SparkSession
+
+from feathub.common.exceptions import FeathubException
+from feathub.dsl.expr_parser import ExprParser
+from feathub.feature_tables.feature_table import FeatureTable
+from feathub.feature_views.feature_view import FeatureView
+from feathub.processors.processor import Processor
+from feathub.processors.spark.ast_evaluator.spark_ast_evaluator import SparkAstEvaluator
+from feathub.processors.spark.dataframe_builder.spark_dataframe_builder import (
+    SparkDataFrameBuilder,
+)
+from feathub.processors.spark.spark_job import SparkJob
+from feathub.processors.spark.spark_processor_config import (
+    SparkProcessorConfig,
+    MASTER_CONFIG,
+)
+from feathub.processors.spark.spark_table import SparkTable
+from feathub.registries.registry import Registry
+from feathub.table.table_descriptor import TableDescriptor
+
+
+class SparkProcessor(Processor):
+    """
+    The SparkProcessor does feature ETL using Spark as the processing engine.
+
+    In the following we describe the keys accepted by the `config` dict passed to the
+    SparkProcessor constructor.
+
+    master: The Spark master URL to connect to.
+    """
+
+    def __init__(self, props: Dict, registry: Registry):
+        """
+        Instantiate the SparkProcessor.
+
+        :param props: The processor properties.
+        :param registry: An entity registry.
+        """
+        super().__init__()
+        self.registry = registry
+
+        config = SparkProcessorConfig(props)
+        spark_builder = SparkSession.builder
+
+        if config.get(MASTER_CONFIG) is not None:
+            spark_builder = spark_builder.master(config.get(MASTER_CONFIG))
+
+        self.spark = spark_builder.getOrCreate()
+        self.dataframe_builder = SparkDataFrameBuilder(self.spark, self.registry)
+        self.executor = ThreadPoolExecutor()
+
+        self.parser = ExprParser()
```

Comment:
Do we need this variable?

Same for other variables defined in this class.

By design we should be able to use SparkProcessor only via those APIs defined in `Processor`.